railties/test/generators/devcontainer_generator_test.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "generators/generators_test_helper"
+require "rails/generators/rails/devcontainer/devcontainer_generator"
+
+module Rails
+  module Generators
+    class DevContainerGeneratorTest < Rails::Generators::TestCase
+      include GeneratorsTestHelper
+
+      tests DevcontainerGenerator
+
+      def test_generates_devcontainer_files
+        run_generator
+
+        assert_file ".devcontainer/compose.yaml"
+        assert_file ".devcontainer/Dockerfile"
+        assert_file ".devcontainer/devcontainer.json"
+      end
+
+      def test_devcontainer_json_no_mounts_by_default
+        run_generator
+
+        assert_devcontainer_json_file do |devcontainer_config|
+          assert_nil devcontainer_config["mounts"]
+        end
+      end
+
+      def test_dev_flag_adds_rails_repo_mounts
+        run_generator ["--dev"]
+
+        assert_devcontainer_json_file do |devcontainer_config|
+          mounts = devcontainer_config["mounts"].sole
+
+          assert_equal "bind", mounts["type"]
+          assert_equal Rails::Generators::RAILS_DEV_PATH, mounts["source"]
+          assert_equal Rails::Generators::RAILS_DEV_PATH, mounts["target"]
```

Comment:
Take my comments with a grain of salt as I am not familiar with the testing codebase and best practices, but it seems like using constants like `Rails::Generators::RAILS_DEV_PATH` inside tests relies on internal behavior which might not be very kosher. Replacing the constant with an path would allow testing only external behavior. Maybe some tests in [`app_generator_test.rb`](https://github.com/rails/rails/blob/main/railties/test/generators/app_generator_test.rb) could be helpful as a reference.

Reply:
yeah, agreed, I think i saw it in an app generator test, but I don't like it, will change.